<H1><CENTER>STATISTICS(ASSIGNMENTS 1,2,3,4)</CENTER></H1>

***

# STATISTICS 1

***

### Problem Statement 1: 
You survey households in your area to find the average rent they are paying.
Find the standard deviation from the following data: 
1550, 1700, 900, 850, 1000, 950.  

 
 

In [1]:
import numpy as np
import pandas as pd
import math
import statistics as stat

In [2]:
l=[1550, 1700, 900, 850, 1000, 950]
mean=sum(l)/len(l)
print("Mean= ",mean)

Mean=  1158.3333333333333


In [3]:
stdev=math.sqrt(sum((x - mean)**2 for x in l) / (len(l)-1))
print("Standard deviation= ",stdev)

Standard deviation=  367.99003609699366


In [4]:
#uisng function
print("Standard deviation = ",np.std(l, ddof=1))

Standard deviation =  367.99003609699366


In [5]:
#using stats
print("Standard deviation= ",stat.stdev(l))

Standard deviation=  367.9900360969936


### Problem 2
Find the variance for the following set of data representing trees in California (heights in feet):
3, 21, 98, 203, 17, 9 
 
 

In [6]:
data=[3, 21, 98, 203, 17, 9]
mean=sum(data)/len(data)
variance=sum((x - mean)**2 for x in data) / (len(data)-1)
print("Variance = ",variance)

Variance =  6219.9


In [7]:
#using function
print("Variance= ",np.var(data, ddof=1))

Variance=  6219.9


In [8]:
#using stats
print("Variance= ",stat.variance(data))

Variance=  6219.9


### Problem 3
In a class on 100 students, 80 students passed in all subjects, 10 failed in one subject, 7 failed in two subjects and 3 failed in three subjects. Find the probability distribution of the variable for number of subjects a student from the given class has failed in. 
 
 

In [9]:
total_students = 100
def prob(failed_students):
    P = failed_students/total_students
    return ("Probability of failed_students = "+str(P))


In [10]:
prob(80)

'Probability of failed_students = 0.8'

In [11]:
prob(10)

'Probability of failed_students = 0.1'

In [12]:
prob(7)

'Probability of failed_students = 0.07'

In [13]:
prob(3)

'Probability of failed_students = 0.03'

***

# STATISTICS 2

***

### Problem 1
A test is conducted which is consisting of 20 MCQs (multiple choices questions) with every MCQ having its four options out of which only one is correct. Determine the probability that a person undertaking that test has answered exactly 5 questions wrong.

In [14]:
#Binomial Distribution B(n,p) is equal to the value k, where k = 0, 1,....,n , is given by 
#nCk * p^k * q^(n-k)
#B(n,p)=(n! / (n-k!)*k!) * (p**k) * (q**(n-k))

B(n,p)=(n! / (n-k!)*k!) * (p**k) * (q**(n-k))

Here, n = 20, k = 20 - 5 = 15, n - k = 20-15 = 5

Here the probability of success = probability of giving a right answer = p = 1/4

Hence, the probability of failure = probability of giving a wrong answer = 1 - p
= 1 - 1/4 = 3/4

When we substitute these values in the formula for Binomial distribution we get,

So, P (exactly 5 out of 20 answers incorrect) = C (20, 5) * (1/4) 15 * (3/4) 5



In [15]:
#Probability,P =((20!)/(15! * 5!)) * (1/4) **15 * (3/4) **5
P= format(((20*19*18*17*16)/(5*4*3*2*1))*((1/4)**15) *((3/4)**5),'0.9f')
print("Probability= ",P)

Probability=  0.000003426


In [16]:
#Another method
import scipy
from scipy.special import comb

n = 20
probofsuccess=(1/4)
proboffailure = 1 - probofsuccess
#Exact 5 failures
k = 20 - 5 
combinations = scipy.special.comb(n, k, exact=False, repetition=False)
probability = format(combinations * np.power((probofsuccess),k)*((proboffailure)**(n-k)),'0.9f')
print("Probability= ",probability)

Probability=  0.000003426


### Problem 2 


A die marked A to E is rolled 50 times. Find the probability of getting a “D” exactly 5 times. 

Here, n = 50, k = 5, n - k = 45.

The probability of success = probability of getting a “D”= p = 1/5

Hence, the probability of failure = probability of not getting a “D” =q= 1 - p = 4/5.

In [17]:
import scipy
from scipy.special import comb
n = 50
k = 5
probofsuccess=1/5
proboffailure = 1 - probofsuccess
combinations = scipy.special.comb(n, k, exact=False, repetition=False)
probability = format(combinations * np.power((probofsuccess),k)*((proboffailure)**(n-k)),'0.9f')
print("Probability= ",probability)

Probability=  0.029531204


### Problem 3
Two balls are drawn at random in succession without replacement from an urn containing 4 red balls and 6 black balls.Find the probabilities of all the possible outcomes. 

In [18]:
Total_balls=6+4

#Probabilities of all possible outcomes

RR=(4/10)*(3/9)
print("Probability_of_two_reds(RR)={:.3f}".format(RR))

RB=(4/10)*(6/9)
print("\nProbability_of_one_red_one_black(RB)={:.3f}".format(RB))

BR=(6/10)*(4/9)
print("\nProbability_of_one_black_one_red(BR)={:.3f}".format(BR))

BB=(6/10)*(5/9)
print("\nProbability_of_two_blacks(BB)={:.3f}".format(BB))

Sum_of_all_probabilities=RR + RB + BR +BB
print("\nSum_of_all_probabilities=",Sum_of_all_probabilities)

Probability_of_two_reds(RR)=0.133

Probability_of_one_red_one_black(RB)=0.267

Probability_of_one_black_one_red(BR)=0.267

Probability_of_two_blacks(BB)=0.333

Sum_of_all_probabilities= 1.0


***

# STATISTICS 3

***

### Problem 1:
Blood glucose levels for obese patients have a mean of 100 with a standard deviation of  15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an effect or not. 
 
 

#### z=(X-μ)/σ

In [19]:
import numpy as np
from scipy import stats
def Zscore(x, mean, stdev):
    return (x - mean)/stdev


x = 108
mean = 100
stdev = 15/np.sqrt(36)

z = Zscore(x, mean, stdev)
print("𝜇= "+str(mean)+"\tX̄= "+str(x)+ "\t𝜎 =  "+str(stdev))
print("z= ",z)
p= stats.norm.cdf(z)
print("p-value associated with 3.20 is ",p)

𝜇= 100	X̄= 108	𝜎 =  2.5
z=  3.2
p-value associated with 3.20 is  0.9993128620620841


#### Performing Hypothesis Testing
Step1:Develop a Hypothesis 
            The population mean is 100.
    
H0: μ = 100
H1: μ ≠ 100
    
Step2:  State significance level:  
        Significance level (α). It is not given in the problem and hence taking it as 5% (0.05).    
    α=0.05
    
Step3: Identify Test Statistic: 
    𝜇=100
    𝜎 = 15
    X̄=108
    n=36
    
    
 z= (108-100) / (15/√36)=3.20
 
Step4: Determine P-value:
         The probability by looking at z- table and p-value associated with 3.20 is 0.9993. 
            Probability of having p-value less than 108 is 0.9993 and more than or equals to 108 is (1-0.9993)=0.0007.


Step5: Compare P-value to Significance level:
        0.0007<0.05.
        Hence  Reject H0,i.e. there is raw cornstarch effect.

### Problem Statement 2:

In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple random sample of 100 voters are surveyed from each state. What is the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state?

In [20]:
import numpy as np
from statsmodels.stats.proportion import proportions_ztest
counts = np.array([52,47])
nobs = np.array([100, 100])

zscore, pvalue = proportions_ztest(counts, nobs, alternative = 'larger')
print('zscore = {:.3f}, pvalue = {:.3f}'.format(zscore, pvalue))
print("The probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is 0.24.")

zscore = 0.707, pvalue = 0.240
The probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is 0.24.


####  Alternate method
n1P1 = 100 * 0.52 = 52,

n1(1 - P1) = 100 * 0.48 = 48,

n2P2 = 100 * 0.47 = 47

n2(1 - P2) = 100 * 0.53 = 53 are each greater than 10, the sample size is large enough.

 
Finding the mean of the difference in sample proportions:

E(p1 - p2) = P1 - P2 = 0.52 - 0.47 = 0.05.
 

Find the standard deviation of the difference.

σd =  sqrt{ [ P1(1 - P1) / n1 ] + [ P2(1 - P2) / n2 ] } 

σd = sqrt{ [ (0.52)(0.48) / 100 ] + [ (0.47)(0.53) / 100 ] } 

σd = sqrt (0.002496 + 0.002491) = sqrt(0.004987) = 0.0706

Finding the probability--

This problem requires us to find the probability that p1 is less than p2. 
This is equivalent to finding the probability that p1 - p2 is less than zero. 
To find this probability, we need to transform the random variable (p1 - p2) into a z-score. 

z = (x - μ  ) / σd = = (0 - 0.05)/0.0706 = -0.7082
z-score being -0.7082 or less is 0.24.


Therefore, the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is 0.24.

### Problem Statement 3:
You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard deviation is 209. How well did you score on the test compared to the average test taker?

### z=(X-μ)/σ
z=(1100 – 1026) / 209 =0.354. 

A z-score of 0.354 is 0.1368 + 0.5= .6368 or 63.68%.

The z-table shown has scores for the RIGHT of the mean. Therefore, we have to add 0.5 for all of the area LEFT of the mean. 



In [21]:
#Alternate method Using stats
import numpy as np
from scipy import stats
def Zscore(x, mean, stdev):
    return (x - mean)/stdev


x = 1100
mean = 1026
stdev = 209

z = Zscore(x, mean, stdev)
#The z-table shown has scores for the RIGHT of the mean. Therefore, we have to add 0.5 for all of the area LEFT of the mean.

print("𝜇= "+str(mean)+"\tX̄= "+str(x)+ "\t  𝜎 =  "+str(stdev))
print("z={:.3f}".format(z))
p= stats.norm.cdf(z)*100
print("p-value associated with z-value {:.3f}".format(z)+" is {:.3f} ".format(p))
confidence_interval =[x-(z*(mean/stdev)),x+(z*(mean/stdev))]
print("My Score is in the range {} - {}  with a  percentage of test-takers scored below me is {:.2f}%".format(mean - stdev,mean + stdev,p))

𝜇= 1026	X̄= 1100	  𝜎 =  209
z=0.354
p-value associated with z-value 0.354 is 63.836 
My Score is in the range 817 - 1235  with a  percentage of test-takers scored below me is 63.84%


***

# STATISTICS 4

***

### Problem Statement 1:
Is gender independent of education level? A random sample of 395 people were
surveyed and each person was asked to report the highest education level they
obtained. The data that resulted from the survey is summarized in the following table:

            High School Bachelors Masters Ph.d. Total
     Female         60     54        46    41    201
      Male          40     44        53    57    194
      Total         100    98        99    98    395

Question: Are gender and education level dependent at 5% level of significance? In
other words, given the data collected above, is there a relationship between the
gender of an individual and the level of education that they have obtained?

#### Chi-square Test Statistic,χ2= Σ  [(O-E)**2] / E ;

where O is the Observed Frequency and E is the Expected Frequency under the null hypothesis

#### E=(Row Total * Column Total)/Sample Size

In [22]:
import numpy as np
import pandas as pd
df = pd.DataFrame(np.array([[60,54,46,41,201], [40,44,53,57,194],[100,98,99,98,395]])
                          ,columns=['HighSchool','Bachelors','Masters','PhD','Row_Total'])
df.rename(index={0:'Female',1:'Male',2:'Col_Total'}, inplace=True)
df

HighSchool  Bachelors  Masters  PhD  Row_Total
Female             60         54       46   41        201
Male               40         44       53   57        194
Col_Total         100         98       99   98        395

In [23]:
expected =  np.outer(df["Row_Total"][0:2],
                     df.loc["Col_Total"][0:4]) / 395

expected = pd.DataFrame(expected)

expected.columns = ['HighSchool','Bachelors','Masters','PhD']
expected.index = ["Female","Male"]

expected

HighSchool  Bachelors    Masters        PhD
Female   50.886076  49.868354  50.377215  49.868354
Male     49.113924  48.131646  48.622785  48.131646

In [24]:
observed = df.iloc[0:2,0:4]
chi_squared_stat = (((observed-expected)**2)/expected).sum().sum()
 #   We call .sum() twice: once to get the column sums and a second time to add the column sums together,
#              returning the sum of the entire 2D table.
print("Chi-Square Statistic= {:.3f}".format(chi_squared_stat))
import scipy.stats as stats
crit = stats.chi2.ppf(q = 0.95, # Find the critical value for 95% confidence*
                      df = 3)   

print("Critical value= {:.3f}".format(crit))

p_value = 1 - stats.chi2.cdf(x=chi_squared_stat,  # Find the p-value
                             df=3)
print("P value= {:.3f}".format(p_value))
print("\n  The critical value of χ2 with 3 degree of freedom is 7.815. Since 8.006 > 7.815, therefore we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance.")

Chi-Square Statistic= 8.006
Critical value= 7.815
P value= 0.046

  The critical value of χ2 with 3 degree of freedom is 7.815. Since 8.006 > 7.815, therefore we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance.


#### An alternate method to find the Critical value,p-vale and degrees of freedom using stats is as follows:


In [25]:
stats.chi2_contingency(observed= observed)

(8.006066246262538,
 0.045886500891747214,
 3,
 array([[50.88607595, 49.86835443, 50.37721519, 49.86835443],
        [49.11392405, 48.13164557, 48.62278481, 48.13164557]]))

The above output shows the chi-square statistic, the p-value and the degrees of freedom followed by the expected counts.The critical value of χ2 with 3 degree of freedom is 7.815. Since 8.006 > 7.815, therefore we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance.

### Problem Statement 2: 
 
Using the following data, perform a oneway analysis of variance using α=.05. Write up the results in APA format. 
 
 
[Group1: 51, 45, 33, 45, 67]  
[Group2: 23, 43, 23, 43, 45]  
[Group3: 56, 76, 74, 87, 56] 

In [26]:
data = {'Group1':[51,45,33,45,67], 
        'Group2':[23,43,23,43,45], 
        'Group3':[56,76,74,87,56]} 
data = pd.DataFrame(data)
data

Group1  Group2  Group3
0      51      23      56
1      45      43      76
2      33      23      74
3      45      43      87
4      67      45      56

#### Step 1: Calculate all the means

In [27]:
means=data.mean()
print("Means of the groups:\n",means)
grand_mean=means.mean()
print("Grand mean= ",round(grand_mean,2))

Means of the groups:
 Group1    48.2
Group2    35.4
Group3    69.8
dtype: float64
Grand mean=  51.13


#### Step 2: Specify the rejection area
α = 0.05
Rejection criteria: K(0.05) < F
This means that if the critical value of F from tables is less than the calculated value of F, we reject the null hypothesis

#### Step 3: Calculate the Sum of Squares
The formula for sum of squares is 
SS(total)=SS(between)+SS(within)

In [28]:
data['g1_sst']=(data.Group1-grand_mean)**2
data['g2_sst']=(data.Group2-grand_mean)**2
data['g3_sst']=(data.Group3-grand_mean)**2
data

Group1  Group2  Group3      g1_sst      g2_sst       g3_sst
0      51      23      56    0.017778  791.484444    23.684444
1      45      43      76   37.617778   66.151111   618.351111
2      33      23      74  328.817778  791.484444   522.884444
3      45      43      87   37.617778   66.151111  1286.417778
4      67      45      56  251.751111   37.617778    23.684444

In [29]:
SST=data.g1_sst.sum()+data.g2_sst.sum()+data.g3_sst.sum()
print("Total Sum of Squares, SS(Total)= ",SST)

Total Sum of Squares, SS(Total)=  4883.733333333334


In [30]:
data['g1_ssw']=(data.Group1-data.Group1.mean())**2
data['g2_ssw']=(data.Group2-data.Group2.mean())**2
data['g3_ssw']=(data.Group3-data.Group3.mean())**2
data

Group1  Group2  Group3      g1_sst      g2_sst       g3_sst  g1_ssw  \
0      51      23      56    0.017778  791.484444    23.684444    7.84   
1      45      43      76   37.617778   66.151111   618.351111   10.24   
2      33      23      74  328.817778  791.484444   522.884444  231.04   
3      45      43      87   37.617778   66.151111  1286.417778   10.24   
4      67      45      56  251.751111   37.617778    23.684444  353.44   

   g2_ssw  g3_ssw  
0  153.76  190.44  
1   57.76   38.44  
2  153.76   17.64  
3   57.76  295.84  
4   92.16  190.44

In [31]:
ssw=data.g1_ssw.sum()+data.g2_ssw.sum()+data.g3_ssw.sum()
print("Sum of Squares within groups, SSW= ",ssw)

Sum of Squares within groups, SSW=  1860.8


In [32]:
data['g1_ssb']=(data.Group1.mean()-grand_mean)**2
data['g2_ssb']=(data.Group2.mean()-grand_mean)**2
data['g3_ssb']=(data.Group3.mean()-grand_mean)**2
data

Group1  Group2  Group3      g1_sst      g2_sst       g3_sst  g1_ssw  \
0      51      23      56    0.017778  791.484444    23.684444    7.84   
1      45      43      76   37.617778   66.151111   618.351111   10.24   
2      33      23      74  328.817778  791.484444   522.884444  231.04   
3      45      43      87   37.617778   66.151111  1286.417778   10.24   
4      67      45      56  251.751111   37.617778    23.684444  353.44   

   g2_ssw  g3_ssw    g1_ssb      g2_ssb      g3_ssb  
0  153.76  190.44  8.604444  247.537778  348.444444  
1   57.76   38.44  8.604444  247.537778  348.444444  
2  153.76   17.64  8.604444  247.537778  348.444444  
3   57.76  295.84  8.604444  247.537778  348.444444  
4   92.16  190.44  8.604444  247.537778  348.444444

In [33]:
ssb=data.g1_ssb.sum()+data.g2_ssb.sum()+data.g3_ssb.sum()
print("Sum of Squares between groups, SSB= ",ssb)

Sum of Squares between groups, SSB=  3022.933333333333


In [34]:
sst=ssb+ssw
print(sst)

4883.733333333333


In [35]:
print("Sum of Squares within groups, SSW= ",ssw)
print("Sum of Squares between groups, SSB= ",ssb)
print("Total Sum of Squares, SST= ",sst)

Sum of Squares within groups, SSW=  1860.8
Sum of Squares between groups, SSB=  3022.933333333333
Total Sum of Squares, SST=  4883.733333333333


#### Step 4: Calculate the Degrees of Freedom

In [36]:
n=15 #total number of samples
k=3 #total number of groups

dft=n-1
dfw =n-k 
dfb =k-1
print("df(total)= "+str(dft)+"\tdf(within)= "+str(dfw)+"\tdf(between)= "+str(dfb))

df(total)= 14	df(within)= 12	df(between)= 2


#### Step 5: Calculate the Mean Squares
Mean Square(between)=SS(between)/dfbetween
Mean Square(within)=SS(within)/dfwithin

In [37]:
MSbetween=ssb/dfb
MSwithin=ssw/dfw
print("Mean Square Between, MS(between)= "+str(MSbetween)+"\nMean Square Within,MS(within)= "+str(MSwithin))

Mean Square Between, MS(between)= 1511.4666666666665
Mean Square Within,MS(within)= 155.06666666666666


#### Step 6: Calculate the F Statistic
F=MS(between)/MS(within)

In [38]:
F=MSbetween/MSwithin
print("F statisitc = ",F)

F statisitc =  9.747205503009457


#### Step 7: Looking up F from table and stating Conclusion

From the table of F distribution, the critical value of F for 0.05 significance and degrees of freedom of(df1 = 12 and df2 = 2) we have:

F = 3.89

Since the calculated(absolute value) of F is greater than the tabulated value, we reject the null hypothesis and conclude that at least two of the means are significantly different from each other.

Effect size

η2=SSB/SST=3022.9/4883.7=0.62

#### APA writeup

F(2, 12)=9.75, p <0.05, η2=0.62.

***

### Problem Statement 3: 
 
Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25. 
 
For 10, 20, 30, 40, 50: 

In [39]:
import numpy as np
import pandas as pd
set1=np.array([10,20,30,40,50])
set2=np.array([5,10,15,20,25])
print("Set1: ",set1)
print("Set2: ",set2)

Set1:  [10 20 30 40 50]
Set2:  [ 5 10 15 20 25]


In [40]:
mean1=set1.mean()
mean2=set2.mean()
print("Mean of set1: ",mean1)
print("Mean of set2: ",mean2)
stdev1=np.std(set1,ddof=1)
stdev2=np.std(set2,ddof=1)
print("\nStandard Deviation of set1: ",stdev1)
print("Standard Deviation of set2: ",stdev2)
var1=(stdev1)**2
var2=(stdev2)**2
print(" \nVariance of set1: ",var1)
print("Variance of set2: ",var2)
F_test=var1/var2
print("\nF_Test= Variance of set1/Variance of set2 = 250/62.5")
print("F Test value is ",F_test)

Mean of set1:  30.0
Mean of set2:  15.0

Standard Deviation of set1:  15.811388300841896
Standard Deviation of set2:  7.905694150420948
 
Variance of set1:  250.0
Variance of set2:  62.5

F_Test= Variance of set1/Variance of set2 = 250/62.5
F Test value is  4.0


***